kubernetes/resource_kubernetes_secret.go

```diff
@@ -22,6 +22,29 @@ func resourceKubernetesSecret() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+			if diff.Id() == "" {
+				return nil
+			}
+
+			// ForceNew if immutable has been set to true
```

Comment:
@DrFaust92 I've changed the logic here slightly. Before this was only doing a ForceNew on changes to the `immutable` attribute itself (and the tests were only testing this). What we want is to ForceNew on the `data`, `binary_data`, and `immutable` attributes _if_ `immutable` has been previously set to true. 

I've also updated the acceptance test to modify the data field to test this.

Reply:
Thanks @jrhouston !